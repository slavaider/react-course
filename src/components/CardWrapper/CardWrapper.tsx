import React from 'react';
import Card from '../Card';

const CardWrapper: React.FC = () => {
  const cardList = new Array(10);
  cardList.fill(0);

  return (
    <div className="row">
      {cardList.map((_, index) => {
        return <Card key={`card-${index}`} name={`card-${index}`} />;
      })}
    </div>
  );
};

export default CardWrapper;
