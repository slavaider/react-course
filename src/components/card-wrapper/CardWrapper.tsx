import React from 'react';
import './CardWrapper.scss';
import Card from '../card/Card';
import { User } from '../../interfaces';

type CardProps = {
  items: User[];
};

const CardWrapper: React.FC<CardProps> = ({ items }: CardProps) => {
  return (
    <div className="CardWrapper row">
      {items.map((user, index) => {
        return <Card key={`card-${index}`} user={user} />;
      })}
    </div>
  );
};

export default CardWrapper;
