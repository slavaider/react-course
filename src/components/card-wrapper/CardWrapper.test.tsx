import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import News from '../../interfaces/news';
import CardWrapper from './CardWrapper';

test('should render', () => {
  const items: News[] = [
    {
      author: '',
      content: '',
      description: '',
      publishedAt: '',
      source: {},
      title: '',
      url: '',
      urlToImage: '',
    },
    {
      author: '',
      content: '',
      description: '',
      publishedAt: '',
      source: {},
      title: '',
      url: '',
      urlToImage: '',
    },
  ];

  const wrapper = render(
    <Router>
      <CardWrapper items={items} />
    </Router>,
  );

  expect(wrapper).toMatchSnapshot();
});
