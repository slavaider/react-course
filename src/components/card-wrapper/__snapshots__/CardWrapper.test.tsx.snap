// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CardWrapper row"
      >
        <div
          class="Card col-xs-12 col-md-3"
        >
          <div
            class="card bg-dark text-white mx-1 my-1"
          >
            <img
              alt=""
              aria-label="image"
              class="card-img-top"
              src=""
            />
            <div
              class="card-body"
            >
              <p
                aria-label="title"
                class="card-text"
              >
                Title: 
                
              </p>
              <p
                aria-label="description"
                class="card-text"
              >
                Description: 
                
              </p>
              <p
                aria-label="publishedAt"
                class="card-text"
              >
                Published: 
                
              </p>
              <p
                aria-label="author"
                class="card-text"
              >
                Author: 
                
              </p>
            </div>
            <div
              class="card-footer"
            >
              <a
                class="card-link"
                href=""
                rel="noreferrer"
                target="_blank"
              >
                Source
              </a>
              <a
                class="card-link ml-2"
                href="/details/"
              >
                Details
              </a>
            </div>
          </div>
        </div>
        <div
          class="Card col-xs-12 col-md-3"
        >
          <div
            class="card bg-dark text-white mx-1 my-1"
          >
            <img
              alt=""
              aria-label="image"
              class="card-img-top"
              src=""
            />
            <div
              class="card-body"
            >
              <p
                aria-label="title"
                class="card-text"
              >
                Title: 
                
              </p>
              <p
                aria-label="description"
                class="card-text"
              >
                Description: 
                
              </p>
              <p
                aria-label="publishedAt"
                class="card-text"
              >
                Published: 
                
              </p>
              <p
                aria-label="author"
                class="card-text"
              >
                Author: 
                
              </p>
            </div>
            <div
              class="card-footer"
            >
              <a
                class="card-link"
                href=""
                rel="noreferrer"
                target="_blank"
              >
                Source
              </a>
              <a
                class="card-link ml-2"
                href="/details/"
              >
                Details
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CardWrapper row"
    >
      <div
        class="Card col-xs-12 col-md-3"
      >
        <div
          class="card bg-dark text-white mx-1 my-1"
        >
          <img
            alt=""
            aria-label="image"
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <p
              aria-label="title"
              class="card-text"
            >
              Title: 
              
            </p>
            <p
              aria-label="description"
              class="card-text"
            >
              Description: 
              
            </p>
            <p
              aria-label="publishedAt"
              class="card-text"
            >
              Published: 
              
            </p>
            <p
              aria-label="author"
              class="card-text"
            >
              Author: 
              
            </p>
          </div>
          <div
            class="card-footer"
          >
            <a
              class="card-link"
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Source
            </a>
            <a
              class="card-link ml-2"
              href="/details/"
            >
              Details
            </a>
          </div>
        </div>
      </div>
      <div
        class="Card col-xs-12 col-md-3"
      >
        <div
          class="card bg-dark text-white mx-1 my-1"
        >
          <img
            alt=""
            aria-label="image"
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <p
              aria-label="title"
              class="card-text"
            >
              Title: 
              
            </p>
            <p
              aria-label="description"
              class="card-text"
            >
              Description: 
              
            </p>
            <p
              aria-label="publishedAt"
              class="card-text"
            >
              Published: 
              
            </p>
            <p
              aria-label="author"
              class="card-text"
            >
              Author: 
              
            </p>
          </div>
          <div
            class="card-footer"
          >
            <a
              class="card-link"
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Source
            </a>
            <a
              class="card-link ml-2"
              href="/details/"
            >
              Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
